targets/linux/distro_handler.go

```diff
@@ -90,30 +90,62 @@ func HandleContainer(c DistroConfig) gwclient.BuildFunc {
 				return nil, nil, err
 			}
 
-			pg := dalec.ProgressGroup(spec.Name)
-			pc := dalec.Platform(platform)
+			var opts []llb.ConstraintsOpt
+			opts = append(opts, dalec.ProgressGroup(spec.Name))
+			opts = append(opts, dalec.Platform(platform))
 
-			worker, err := c.Worker(sOpt, pg, pc)
+			worker, err := c.Worker(sOpt, opts...)
 			if err != nil {
 				return nil, nil, err
 			}
 
-			deb, err := c.BuildPkg(ctx, client, worker, sOpt, spec, targetKey, pg, pc)
-			if err != nil {
-				return nil, nil, err
+			// Find pre-built package from build context.
+			var pkgSt llb.State
+			var foundPrebuiltPkg bool
+
+			// Pre-built packages can be provided via the build context by providing one of these package name formats:
+			// 1. {targetKey}-pkg - Target specific package context (e.g. rpm-pkg, deb-pkg).
+			// 2. pkg - Generic package for any package type.
+			//
+			// Target specific package will take precedence over the generic package.
+
+			// Try checking for target specific package first.
+			targetSpecificName := targetKey + dalec.PreBuiltPkgSuffix
+			targetPkgSt, err := sOpt.GetContext(targetSpecificName, dalec.WithConstraints(opts...))
+			if err == nil && targetPkgSt != nil {
+				pkgSt = *targetPkgSt
+				foundPrebuiltPkg = true
+				opts = append(opts, dalec.ProgressGroup(fmt.Sprintf("Using pre-built package from %s context", targetSpecificName)))
+			} else {
+				// Try generic package.
+				genericPkgSt, err := sOpt.GetContext(dalec.GenericPkg, dalec.WithConstraints(opts...))
+				if err == nil && genericPkgSt != nil {
+					pkgSt = *genericPkgSt
+					foundPrebuiltPkg = true
+					opts = append(opts, dalec.ProgressGroup(fmt.Sprintf("Using pre-built package from %s context", dalec.GenericPkg)))
```

Comment:
I think these should be warnings we write to the client instead of progress groups.
`client` has a `Warn` function and, IIRC we have a helper method in the `frontend` package to simplify that a little.

Reply:
done

Reply:
Thanks for highlighting this btw, I did not know about this function.